### DISCUSSION TITLE
NOTE: (for people who cannot install Chrome extensions (i.e. on a school chromebook))

### DISCUSSION THREAD

**Discussion by RuralAnemone** [LibreScore/dl-librescore#2] [Announcements]
If you cannot install tampermonkey on your device, just make a bookmarklet with this code as the "link":

`javascript:(function () {let code = document.createElement('script');code.src = 'https://github.com/LibreScore/dl-librescore/releases/latest/download/dl-librescore.user.js';document.body.appendChild(code);}())`

**Comment by RuralAnemone** [comment]
or you can also use Xmader's earlier version if you need other file types:
`javascript:(function () {let code = document.createElement('script');code.src = 'https://msdl.librescore.org/install.user.js';document.body.appendChild(code);}())`
although I'll probably get cancelled for saying this...

**Comment by sealsrock12** [comment]
Xmader's earlier version relies on a dataset that doesn't exist anymore, so unless someone else is hosting those particular IPFS urls, the dataset won't work for the vast majority of scores.

Not sure how you are supposed to use MSCZ/XML files on school chromebooks, though.

**Comment by RuralAnemone** [comment]
true
although I've never had any issues with it so idk


lol yeah good point with the XML files :)

**Comment by sealsrock12** [comment]
That's very interesting, the top 3 scores I checked don't work with the old script. Could you share a score that does still work?

**Comment by RuralAnemone** [comment]
https://musescore.com/arcandius/mother-3-033-gentle-rain

video: https://ruralanemone.tech/media/video/Untitled_%20Feb%205%202022%2012_44%20PM.mp4

**Comment by PeterNjeim** [comment]
> https://musescore.com/arcandius/mother-3-033-gentle-rain
> 
> video: https://ruralanemone.tech/media/video/Untitled_%20Feb%205%202022%2012_44%20PM.mp4

I don't see you downloading MSCZ in that video

**Comment by RuralAnemone** [comment]
OH sorry I didn't think you meant MSCZ
hang on

**Comment by sealsrock12** [comment]
MIDI/PDF/MP3 don't use the dataset, and use the same method in the old script as in the new one

**Comment by RuralAnemone** [comment]
ah
yeah no MSCZs don't work
I feel like I remember the "open in librescore" button working with the old script fairly recently, but I just checked again and it doesn't

**so yeah use the new script, it's less misleading and all the buttons work**

also what I thought you meant by the MSCZ/XML files was "what are you going to use them for" (not "how will you download them")
sorry for the misunderstanding!

**Comment by PeterNjeim** [comment]
lol, MSCZ files are the most sought after files, that's why the app is so popular. MSCZ files are "mother of all files", capable of being openeed in the MuseScore app and converted to any other file type

**Comment by PeterNjeim** [comment]
> also what I thought you meant by the MSCZ/XML files was "what are you going to use them for" (not "how will you download them") sorry for the misunderstanding!

He was actually saying "how will you use them on a school Chromebook"

**Comment by SuperEpix** [comment]
What do you use this for?

**Reply by PeterNjeim** [reply]
This discussion was about how to run the Userscript on computers that don't permit browser extensions. If you're asking about what the Userscript does, it's explained in the table on the main page: https://github.com/LibreScore/dl-librescore#readme

**Comment by Vessel-01** [comment]
will it work on firefox?

**Reply by sealsrock12** [reply]
The bookmarklet will work on Firefox. But the userscript probably will too, and we suggest using that instead if you can.

**Comment by RiddSarav** [comment]
can someone attach a video of how to use it? Nothing happends when I click the bookmark

**Reply by RuralAnemone** [reply]
the url appears to have changed

I just tried it with the one currently in the readme and it works nicely
I'll edit the script in the original comment

thank you for bringing this to my attention :)
[Screen recording 2023-02-21 9.08.17 AM.webm](https://user-images.githubusercontent.com/64103005/220382884-79b88599-df0d-46fb-9044-b2e65eacf33b.webm)

**Reply by RuralAnemone** [reply]
tl;dr: I edited the original comment to the correct link
